File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/command/receiver.dart
class Receiver {
  final Map<String, int> _items = {};

  void addItem(String item) {
    _items[item] = (_items[item] ?? 0) + 1;
  }

  void removeItem(String item) {
    final cur = _items[item] ?? 0;
    if (cur <= 1) {
      _items.remove(item);
    } else {
      _items[item] = cur - 1;
    }
  }

  void clearItems() {
    _items.clear();
  }

  // helpers
  List<MapEntry<String, int>> list() => _items.entries.toList();
  int count(String item) => _items[item] ?? 0;
  int size() => _items.length;
}

File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/command/receiver.ts
export class Receiver {
  private items = new Map<string, number>();
  addItem(item: string) { this.items.set(item, (this.items.get(item) ?? 0) + 1); }
  removeItem(item: string) {
    const cur = this.items.get(item) ?? 0;
    if (cur <= 1) this.items.delete(item); else this.items.set(item, cur - 1);
  }
  clearItems() { this.items.clear(); }
  list(): Array<{ key: string; qty: number }> { return Array.from(this.items.entries()).map(([key, qty]) => ({ key, qty })); }
}
